### PR TITLE
Support getting the flow definition and schema used to start a run

### PR DESCRIPTION
**Pull request by kurtmckee**
Features
--------

-   Support retrieval of the flow definition and input schema used to start a run.

    This is supported in the new ``flow run-definition`` subcommand.

### CODE REVIEW

File: globus_automate_client/cli/flows.py [hunk 1]
```diff
@@ -898,6 +898,32 @@ def flow_action_status(
     ).run_and_render()
 
 
+@app.command("run-definition")
+def get_flow_definition_for_run(
+    run_id: str = typer.Argument(...),
```
Comment (ada-globus) [inline globus_automate_client/cli/flows.py:903]: "run" = ❤️

File: globus_automate_client/action_client.py [hunk 2]
```diff
@@ -131,6 +131,16 @@ def status(self, action_id: str) -> GlobusHTTPResponse:
 
         return self.get(f"{quote(action_id)}/status")
 
+    def get_definition(self, action_id: str) -> GlobusHTTPResponse:
+        """
+        Query the Action Provider for the status of executed Action
+
+        :param action_id: An identifier that uniquely identifies an Action
+            executed on this Action Provider.
+        """
```
Comment (sirosen) [inline globus_automate_client/action_client.py:140]: I recognize that the class hierarchy here is all kinds of misleading, and that this is both a client for APs and-also the client for Specific Flows.

But what should we put in the docstring here? (Right now looks like a copy-paste)

I'm half-tempted to say that the docstring just says "for Flows only" and nothing more?

**Review by sirosen — APPROVED** [review]
I admit that I largely was comparing against surrounding code to see if things matched up, but by that standard this seems good.

I noticed one docstring which merits attention.

Since this library and CLI is being deprecated anyway, I'm inclined to allow this to move forward with less stringent review than is typical.

Therefore, I'm pre-approving it right now! If you take a look at the docstring issue and modify only that, please consider this approval valid for whatever reasonable change you make there. 👍

File: globus_automate_client/cli/flows.py [hunk 3]
```diff
@@ -898,6 +898,32 @@ def flow_action_status(
...
+    ),
+    flows_endpoint: str = flows_env_var_option,
+    output_format: OutputFormat = output_format_option,
+    verbose: bool = verbosity_option,
+):
+    """
+    Get the flow definition and input schema used to start a run.
+    """
+    fc = create_flows_client(CLIENT_ID, flows_endpoint)
+    method = functools.partial(
```
Comment (ada-globus) [inline globus_automate_client/cli/flows.py:921]: I wouldn't spend more than a handful of minutes on this if it is even possible, but given that the route doesn't require the flow ID nor permissions on the flow itself, do we need to solicit it in this command?

File: globus_automate_client/cli/flows.py [hunk 4]
```diff
@@ -898,6 +898,32 @@ def flow_action_status(
...
+        None,
+        help="The scope this Flow uses to authenticate requests.",
+        callback=url_validator_callback,
+    ),
+    flows_endpoint: str = flows_env_var_option,
+    output_format: OutputFormat = output_format_option,
+    verbose: bool = verbosity_option,
+):
+    """
+    Get the flow definition and input schema used to start a run.
```
Comment (ada-globus) [inline globus_automate_client/cli/flows.py:918]: `<pedantic>`

```suggestion
    Get the flow definition and input schema used to start this run.
```

`</pedantic>`

**Review by ada-globus — APPROVED** [review]
This seems good to me! I noted a couple minor thoughts, but I'm totally fine if this merges as-is.

File: globus_automate_client/cli/flows.py (hunk 3 above)
Comment (kurtmckee) [reply globus_automate_client/cli/flows.py:921]: No, we do not. This same requirement exists on all of the run-related routes, due to how the authorizers are set up (and I'm pretty sure I created a story about removing this requirement, but the priority continues to be to migrate to the Globus CLI).

I already didn't spend more than a handful of minutes on this, which is why this requirement still exists here. 😢😞

File: globus_automate_client/action_client.py (hunk 2 above)
Comment (kurtmckee) [reply globus_automate_client/action_client.py:140]: You're correct, this was copy-and-paste. It took all of two minutes to create IIRC.

I can add some text that says "for Flows only". I'm leery of spending more than two additional minutes on this PR, though, so heads-up that I hold that opinion!

File: globus_automate_client/action_client.py [hunk 5]
```diff
@@ -131,6 +131,16 @@ def status(self, action_id: str) -> GlobusHTTPResponse:
 
         return self.get(f"{quote(action_id)}/status")
 
+    def get_definition(self, action_id: str) -> GlobusHTTPResponse:
+        """
+        Query the Action Provider for the status of executed Action
```
Comment (kurtmckee) [inline globus_automate_client/action_client.py:136]:
```suggestion
        Get the flow definition for a given run ID.
```

**Pull request merged**